unit-tests/internal/internal-tests-extrinsic.cpp

```diff
@@ -271,16 +270,12 @@ TEST_CASE("Extrinsic memory leak detection", "[live]")
                     auto frame_num = f.get_frame_number();
                     auto time_of_arrival = f.get_frame_metadata(RS2_FRAME_METADATA_TIME_OF_ARRIVAL);
 
-                    if (std::find(frame_number[stream_type].begin(), frame_number[stream_type].end(), frame_num) == frame_number[stream_type].end())
+                    if (!new_frame[stream_type])
                     {
-                        if (!new_frame[stream_type])
-                        {
-                            frame_number[stream_type].push_back(frame_num);
-                            streams_delay[stream_type].push_back(time_of_arrival - start_time_milli);
-                            new_frame[stream_type] = true;
-                        }
-                        new_frame[stream_type] += 1;
+                        streams_delay[stream_type].push_back(time_of_arrival - start_time_milli);
+                        new_frame[stream_type] = true;
                     }
+                    new_frame[stream_type] += 1;
```

Comment:
Is this still needed?

Reply:
it is relevant to make sure that at least 20 new frames of each type are received to prevent FW issues (see condition to finish each iteration)